### PR TITLE
Refactor basic-site, add grav

### PR DESCRIPTION
**Pull request by NerdyProjects**
This PR
- refactors basic-site to include letsencrypt as well. The DNS list is automatically generated from server_name + redirects. The redirect nginx config is split into the basic-site, also basic-site can now generate a standalone nginx config via parameter (this is mainly useful to be used as a template nginx config).

For the grav part, a php-fpm role is added that adds the nginx php fpm upstream and a fpm pool. A grav role contains grav specific nginx config and the website checkout.

For the actual grav deployment, there are no best practices regarding the installation of plugins. For now, it is expected that a fully functioning grav lives inside the repository.

### CODE REVIEW

**Comment by nicksellen** [comment]
Looks great! Do you think it's safely runnable now?

File: roles/basic-site/tasks/main.yml [hunk 1]
```diff
@@ -21,4 +21,51 @@
...
+    content: |
+      {% for entry in dns %}
+      {{ entry }}
+      {% endfor %}
+    backup: true
+  notify:
+    - generate cert
+
+- name: nginx config
+  when: setup_html_vhost is defined
```
Comment (nicksellen) [inline roles/basic-site/tasks/main.yml:42]: Nothing seems to actually set this? If it's not used, then I would remove it.

Comment (NerdyProjects) [reply roles/basic-site/tasks/main.yml:42]: I left it in intentionally as a "basic template". But you are right, it is not used anywhere and could be removed...

**Comment by NerdyProjects** [comment]
I used this to setup and deploy grav.kanthaus.online

File: playbooks/foodsharing-taiwan/setup.playbook.yml [hunk 2]
```diff
@@ -4,6 +4,12 @@
   become: yes
   vars_files:
     - secrets.vars.yml
+  vars:
+    site: foodsharing-Taiwan
```
Comment (nicksellen) [inline playbooks/foodsharing-taiwan/setup.playbook.yml:8]: Whoops, the change to an uppercase `Taiwan` meant I just deployed a whole second copy of the site that slightly broke everything :/

Comment (NerdyProjects) [reply playbooks/foodsharing-taiwan/setup.playbook.yml:8]: Oh no :( I am very sorry.

**Pull request merged**